website/app/validators.php

```diff
@@ -24,7 +24,11 @@
 }, _('This language is not supported'));
 
 $validator->addValidator('ciphered_password', function ($field, $input, $param = null) {
-    return substr($input[$field], 0, 7) === '$2a$11$';
+    $expectedPasswordPrefix = sprintf('$2a$%02d$', GK_PASSWORD_HASH_ROTATION);
+    $expectedValidHashRotation = GK_PASSWORD_MIN_HASH_ROTATION <= GK_PASSWORD_HASH_ROTATION && GK_PASSWORD_HASH_ROTATION <= GK_PASSWORD_MAX_HASH_ROTATION;
```

Comment:
This check may now be removed from here as it is present in confgi checks

Reply:
good point (expected! :) 

I were asking myself about that point as healthcheck marks container as _unhealthy_ but container is always running.. so I think we should keep this check to avoid issue.

Reply:
> healthcheck marks container as unhealthy but container is always running..

Hum, is that a docker on Windows specificity? Un-healthy containers are removed from the docker swarm internal load balancer. So we should never access there. But anyway, the overhead is so small that we can keep it, it's just sad to have "the same code" both places :/

Reply:
I agree 

should we hard-add `$config->validate()` at some root point when we make const available ? 
-pro: access const if an only if config is valid in any situation and no more duplicate
-con: add validation step for all gk pages as requirement

Reply:
Personally, I don't think we need this check here at all. The healthcheck should prevent us from having a container running if it's configuration is wrong, I'm a bit surprised when you said "healthcheck marks container as unhealthy but container is always running..." I planned to do some testing tonight about that case ;) For me an unhealthy container should stop processing requests.

Reply:
But don't block this PR for that, it doesn't hurt in itself :)

Reply:
Just checked on a swarm deployment, as soon as the container is `unhealthy` it is killed and re-spawned. That's probably not the case with `docker-compose`

Reply:
thanks for the test

I will remove the double check here because:
- under 10, (ex. 7), I am able to create a password an re-use it without code change
- and up to 99, I got a wrong expectedPasswordCheck, which is ok :)